e2e/test/MessageSendE2EMultiplexingTests.cs

```diff
@@ -242,12 +236,84 @@ public async Task Message_X509_DeviceSendSingleMessage_MuxWithPooling_AmqpWs()
+            int poolSize,
+            int devicesCount,
+            bool useSameDevice = false)
+        {
+            var transportSettings = new ITransportSettings[]
+            {
+                new AmqpTransportSettings(transport)
+                {
+                    AmqpConnectionPoolSettings = new AmqpConnectionPoolSettings()
+                    {
+                        MaxPoolSize = unchecked((uint)poolSize),
+                        Pooling = true
+                    }
+                }
+            };
+
+            ICollection<ModuleClient> moduleClients = new List<ModuleClient>();
+            Dictionary<ModuleClient, int> moduleClientConnectionStatusChangeCount = new Dictionary<ModuleClient, int>();
+
+            try
+            {
+                _log.WriteLine($"{nameof(MessageSendE2EMultiplexingTests)}: Starting the test execution for {devicesCount} modules");
+
+                for (int i = 0; i < devicesCount; i++)
+                {
+                    ConnectionStatus? lastConnectionStatus = null;
+                    ConnectionStatusChangeReason? lastConnectionStatusChangeReason = null;
+                    int setConnectionStatusChangesHandlerCount = 0;
+
+                    string devicePrefix = useSameDevice ? DevicePrefix : $"{DevicePrefix}_{i}_";
+                    string modulePrefix = useSameDevice ? $"{ModulePrefix}_{i}_" : ModulePrefix;
+
+                    TestModule testModule = await TestModule.GetTestModuleAsync(devicePrefix, modulePrefix).ConfigureAwait(false);
+                    ModuleClient moduleClient = ModuleClient.CreateFromConnectionString(testModule.ConnectionString, transportSettings);
+                    moduleClients.Add(moduleClient);
+
+                    moduleClient.SetConnectionStatusChangesHandler((status, statusChangeReason) =>
+                    {
+                        setConnectionStatusChangesHandlerCount++;
+                        lastConnectionStatus = status;
+                        lastConnectionStatusChangeReason = statusChangeReason;
+                        _log.WriteLine($"{nameof(MessageSendE2EMultiplexingTests)}.{nameof(ConnectionStatusChangesHandler)}: status={status} statusChangeReason={statusChangeReason} count={setConnectionStatusChangesHandlerCount}");
+                        moduleClientConnectionStatusChangeCount[moduleClient] = setConnectionStatusChangesHandlerCount;
+                    });
+
+                    _log.WriteLine($"{nameof(MessageSendE2EMultiplexingTests)}: Preparing to send message for module {i}");
+                    await moduleClient.OpenAsync().ConfigureAwait(false);
+                    await MessageSend.SendSingleMessageModuleAndVerifyAsync(moduleClient, testModule.DeviceId).ConfigureAwait(false);
+                }
+            }
+            finally
+            {
+                // Close and dispose all of the module client instances here
+                foreach (ModuleClient moduleClient in moduleClients)
```

Comment:
No non-cleanup actions in the `finally` block. (e.g. calling `CloseAsync` or `Assert`).
Use `using` for dispose as much as possible instead of implementing your own.